axlearn/cloud/gcp/job.py

```diff
@@ -639,6 +639,327 @@ def _execute(self) -> Any:
+            env_vars["NCCL_PROTO"] = "Simple"
+            env_vars["NCCL_DEBUG"] = "WARN"
+            env_vars["NCCL_DEBUG_SUBSYS"] = "INIT,GRAPH,ENV,TUNING,NET,VERSION"
+            env_vars["NCCL_NET_GDR_LEVEL"] = "PIX"
+            env_vars["NCCL_P2P_PXN_LEVEL"] = "0"
+            env_vars["NCCL_GPUDIRECTTCPX_FORCE_ACK"] = "0"
+            env_vars["NCCL_GPUDIRECTTCPX_TX_COMPLETION_NANOSLEEP"] = "1000"
+            env_vars["NCCL_GPUDIRECTTCPX_PROGRAM_FLOW_STEERING_WAIT_MICROS"] = "1000000"
+            env_vars[
+                "NCCL_GPUDIRECTTCPX_TX_BINDINGS"
+            ] = "eth1:8-21,112-125;eth2:8-21,112-125;eth3:60-73,164-177;eth4:60-73,164-177"
+            env_vars[
+                "NCCL_GPUDIRECTTCPX_RX_BINDINGS"
+            ] = "eth1:22-35,124-139;eth2:22-35,124-139;eth3:74-87,178-191;eth4:74-87,178-191"
+            env_vars["NCCL_GPUDIRECTTCPX_SOCKET_IFNAME"] = "eth1,eth2,eth3,eth4"
+            env_vars["NCCL_GPUDIRECTTCPX_CTRL_DEV"] = "eth0"
+            env_vars["NCCL_GPUDIRECTTCPX_UNIX_CLIENT_PREFIX"] = "/run/tcpx"
+            env_vars["NCCL_DYNAMIC_CHUNK_SIZE"] = "524288"
+            env_vars["NCCL_P2P_NET_CHUNKSIZE"] = "524288"
+            env_vars["NCCL_P2P_PCI_CHUNKSIZE"] = "524288"
+            env_vars["NCCL_P2P_NVL_CHUNKSIZE"] = "1048576"
+            env_vars["NCCL_NSOCKS_PERTHREAD"] = "4"
+            env_vars["NCCL_SOCKET_NTHREADS"] = "1"
+            env_vars["NCCL_SOCKET_IFNAME"] = "eth0"
+            env_vars["NCCL_NVLS_ENABLE"] = "0"
+
+        env_vars.update(cfg.env_vars)
+        # K8s expects each env variable to be a dict
+        k8s_env_vars = [{"name": name, "value": value} for name, value in env_vars.items()]
+        k8s_env_vars.append(
+            {
+                "name": "PROCESS_ID",
+                "valueFrom": {
+                    "fieldRef": {
+                        "fieldPath": (
+                            "metadata.annotations['batch.kubernetes.io/" "job-completion-index']"
+                        ),
+                    }
+                },
+            },
+        )
+
+        command = ["bash", "-c", cfg.command]
+        if cfg.accelerator.instance_type.startswith("a3-highgpu"):
+            command.append("touch /run/tcpx/terminated")
+
+        return dict(
+            name=cfg.name,
+            image=self._bundler.id(cfg.name),
+            ports=[
+                dict(containerPort=8080),  # Port for MXLA coordinator.
+            ],
+            securityContext=dict(privileged=True),
+            # TODO(markblee): Improve SIGTERM behavior for command.
+            command=command,
+            resources=dict(limits={"nvidia.com/gpu": "8"}),
+            env=k8s_env_vars,
+            volumeMounts=volume_mounts,
+        )
+
+    def _build_init_container(self) -> Nested[Any]:
+        """Builds a config for a single container."""
+        cfg: GPUGKEJob.Config = self.config
+        if cfg.accelerator.instance_type.startswith("a3-highgpu"):
```

Comment:
I noticed that we have this check in a lot of places. What are the other instance types that we intend to support?

Reply:
I am prepping to also add a3-mega which will require it's own unique setup and configuration sadly. I will think of a way to make it cleaner but we do need machine type specific checks for A3/A3 Mega/ A3 Ultra/ A4

Reply:
I've removed this check for now and will add support for multiple instance type e.g. a3 MEGA at a later time